components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -56,11 +146,27 @@ void EthTxController::AddUnapprovedTransaction(
     mojom::TxDataPtr tx_data,
     const std::string& from,
     AddUnapprovedTransactionCallback callback) {
-  auto tx = EthTransaction::FromTxData(tx_data);
+  FixMissingTxData(&tx_data);
+  if (from.empty()) {
+    std::move(callback).Run(
+        false, "",
+        l10n_util::GetStringUTF8(IDS_WALLET_ETH_SEND_TRANSACTION_FROM_EMPTY));
+    return;
+  }
+  std::string error;
+  if (!EthTxController::ValidateTxData(tx_data, &error)) {
+    std::move(callback).Run(false, "", error);
+    return;
+  }
+  auto tx = EthTransaction::FromTxData(tx_data, false);
```

Comment:
why do we need strict=false?

Reply:
Because for example the renderer can add a transaction and not specify a nonce value, we should still accept it. Same for gas fields and value. This does a best effort to get transaction data information and then validates it later.